relation/index.js

```diff
@@ -108,7 +111,39 @@ module.exports = yeoman.generators.Base.extend({
           }
           return m;
         },
-        validate: validateName
+        validate: function(value) {
+
+            var done = this.async();
+
+            //validfn returns false if relationship 
+            //has the same name with properties
+            var validfn = function(value, cb) {
+
+              //set flag isValid
+              var isValid = true;
+           
+              modelDef.properties(function(err, list) {
+                if(err) return done(err);
+                else {
```

Comment:
Style: use curly braces consistently. Since you use them in `else` block, it's better to use them in `if` block too. Plus add a space after `if` keyword.

```js
if (err) {
  return done(err);
} else {
  // ...
}
```